assets/aml-benchmark/components/src/inference_postprocessor/inference_postprocessor.py

```diff
@@ -420,19 +441,25 @@ def run(self):
             if "processor_order" in self.kwargs
             else None
         )
-        result_df = pd.DataFrame()
-        result_df = self.read_ground_truth_dataset(result_df)
-        result_df = self.extract_inferences(result_df, key, processor_order)
-        if len(result_df) > 0:
-            result_df.to_json(self.result, lines=True, orient="records")
+        actual_df = self.read_ground_truth_dataset()
+        predicted_df = self.extract_inferences(key, processor_order)
+        pd.concat([actual_df, predicted_df], axis=1).to_json(self.result, lines=True, orient='records')
         return
 
-    def run_user_preprocessor(self) -> None:
+    def run_user_postprocessor(self) -> None:
         """Postprocessor run using custom template."""
         try:
+            argss = [
+                "--prediction_dataset",
+                self.prediction_dataset,
+                "--output_dataset",
+                self.result
+            ]
+            if self.ground_truth_dataset:
+                argss.extend("--ground_truth_dataset", self.ground_truth_dataset)

```

Comment:
Please include this in the test. This should have failed, the correct statement should be:
`argss.extend(["--ground_truth_dataset", self.ground_truth_dataset])`

`extend` method accepts one iterable argument.

Reply:
Done. Thanks.